### ISSUE TITLE
Run `bundle install` if Gemfile but no `Gemfile.lock`

### ISSUE THREAD

**Issue by Earlopain** [Shopify/ruby-lsp#1944]
### Reproduction steps

Check out a fresh copy of a repo with `Gemfile.lock` in `.gitignore` (rubocop/rubocop for example).

The LSP starting will first show a notice about mulitroot workspaces and after discarding that refuse to start with the following:

> Project contains a Gemfile, but no Gemfile.lock. Run `bundle install` to lock gems and restart the server

Would it make sense to automatically run `bundle install` for the user in these cases? This would probably require explicit user confirmation, similarly to how VSCode asks about trusted folders. Gems may execute arbitrary commands during installation (I think, especially C extensions during compilation). Perhaps the confirmation of VSCode is already enough, _if_ the extension isn't opting in to trusted mode, https://code.visualstudio.com/docs/editor/workspace-trust. I believe that is the case currently, however looks to be vscode specific and not part of the LSP protocol.

**Comment by andyw8** [comment]
Similarly, this also happens if you use `bundle gem` when creating a new gem, since there's initially no `Gemfile.lock`.

**Comment by vinistock** [comment]
After discussing this with the team, our general sentiment is that the LSP should not run bundle install automatically on a non-locked project. Having the bundle locked is fundamental for anything to work really. Without that, you're unable to run tests, run linters or really do much.

That said, we also believe that the experience is currently quite odd for that scenario. We are only checking if the [lockfile exists](https://github.com/Shopify/ruby-lsp/blob/058059691ae382588290b35f44aac58a65062274/vscode/src/rubyLsp.ts#L121) and then we display a warning including information about multi-root workspaces.

We should definitely include a check to see if a `Gemfile` or `gems.rb` file exists and then we can definitely offer from the VS Code extension to first run bundle install on the project's bundle before trying to launch the language server. How does that sound?

**Comment by vinistock** [comment]
Yes, indeed the point of discussing was the automatic aspect. Because we actually run bundle install only for the custom bundle under `.ruby-lsp`, which would create `.ruby-lsp/Gemfile.lock`, but no top level lockfile - which can be a bit weird.

We should definitely show a message and give the chance for the user to run bundle install if desired.

**Comment by tunacasserole** [comment]
This happens to me using neovim.  My project has a gemfile.lock and it is not gitignored.
